src/components/caption/caption.tsx

```diff
@@ -83,10 +82,16 @@ export class Caption extends Component<ExtendedCaptionProps> {
   }
 
   shouldComponentUpdate(nextProps: ExtendedCaptionProps) {
-    const {indexMap, highlighted, isAutoScrollEnabled, activeSearchIndex, longerThanHour} = this.props;
+    const {indexMap, highlighted, isAutoScrollEnabled, activeSearchIndex, longerThanHour, caption} = this.props;
     if (longerThanHour !== nextProps.longerThanHour) {
       return true;
     }
+    if(Object.keys(highlighted).length === 0){
+      return true;
+    }
+    if(!Object.keys(highlighted).some(c => c === caption.id)) {
+      return false;
```

Comment:
conditions should return `true` only, since if any of them returns `false` next conditions won't be checked.